### ISSUE TITLE
Duplicated elements id in checkout page

### ISSUE THREAD

**Issue by angelo983** [magento/magento2#13415]
### Preconditions
1. Magento 2.1.10

### Steps to reproduce
1. Go to Stores -> Configuration -> Sales -> Sales under Gift Options and set both Allow Gift Messages to YES
2. Put something in cart and go to checkout or cart page

### Expected result
1. Two forms for gift messages having fields with unique ids

### Actual result
1. Two forms for gift messages having fields with non unique ids
```
[DOM] Found 2 elements with non-unique id #gift-message-whole-from: (More info: https://goo.gl/9p2vKq) <input type=​"text" id=​"gift-message-whole-from" class=​"input-text" data-bind=​"value:​ getObservable('sender')​">​ <input type=​"text" id=​"gift-message-whole-from" class=​"input-text" data-bind=​"value:​ getObservable('sender')​">
```
```
[DOM] Found 2 elements with non-unique id #gift-message-whole-to: (More info: https://goo.gl/9p2vKq) <input type=​"text" id=​"gift-message-whole-to" class=​"input-text" data-bind=​"value:​ getObservable('recipient')​">​ <input type=​"text" id=​"gift-message-whole-to" class=​"input-text" data-bind=​"value:​ getObservable('recipient')​">​
```
Non unique ids alert appears too in checkout payment choice:
```
[DOM] Found 2 elements with non-unique id #billing-new-address-form: (More info: https://goo.gl/9p2vKq) <fieldset id=​"billing-new-address-form" class=​"fieldset address">​…​</fieldset>​ <fieldset id=​"billing-new-address-form" class=​"fieldset address">​…​</fieldset>​
```

**Comment by magento-engcom-team** [comment]
@angelo983, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by psuthar12** [comment]
@angelo983  did you got any solution for this ? I am getting this error in Magento 2.1.8 even when both Allow Gift Messages are set to No

**Comment by ishakhsuvarov** [comment]
Similar report: #14126

**Comment by EmilyPepperman** [comment]
We have this error as well for other elements with non-unique id #... running on Magento 2.2.3 and Gift Options and set both Allow Gift Messages to NO

**Comment by ghost** [comment]
Hi @EmilyPepperman I've this similar issue: (in checkout page)

[DOM] Found 2 elements with non-unique id #send2: (More info: https://goo.gl/9p2vKq [^]) 
`<button type=?"submit" class=?"action action-login secondary" name=?"send" id=?"send2">?…?</button>?`

Anyone has experienced same situation?

Thanks,

**Comment by bmzero** [comment]
Same situation for me on a new 2.2.3 install.

**Comment by mariamghalleb** [comment]
any solution ?

**Comment by ghost** [comment]
Also having this problem.

**Comment by zennco** [comment]
Similar issue on  2.2.3 

> [DOM] Found 2 elements with non-unique id #billing-new-address-form:

Noticed that **Cash on delivery** payment method form had same id (#billing-new-address-form) as Authorize.net method.

Solved by disabling the **Cash On Delivery** method.

Magento team should use unique html ids for every method templates.

**Comment by ghost** [comment]
Hi @Deep256 any update on this? thanks!

**Comment by Deep256** [comment]
HI @bruno-serfe.
Yes. I want create a pull-request with update today.

**Comment by ghost** [comment]
Hi @Deep256 will this be ready in Magento v2.3? thanks

**Comment by magento-engcom-team** [comment]
Hi @angelo983. Thank you for your report. 
The issue has been fixed in magento/magento2#15585 by @julienanquetil in 2.2-develop branch
Related commit(s):
 - [2e4bd95d910f577f8999d6a5d7dcbb8f14973e92](https://github.com/magento/magento2/commit/2e4bd95d910f577f8999d6a5d7dcbb8f14973e92)
 - [187c5290e4db5aba1af3d3b6ec53c75b99f52a89](https://github.com/magento/magento2/commit/187c5290e4db5aba1af3d3b6ec53c75b99f52a89)
 - [b90d93dd0c123d5af97d8257024e5a95b3257a25](https://github.com/magento/magento2/commit/b90d93dd0c123d5af97d8257024e5a95b3257a25)

The fix will be available with the upcoming 2.2.6 release.

**Comment by sidolov** [comment]
Hi @angelo983. Thank you for your report. 
The issue has been fixed in magento/magento2#15705 by @julienanquetil in 2.3-develop branch
Related commit(s):
 - [82d0491bc5f697c77092c686038f14f2b60efc34](https://github.com/magento/magento2/commit/82d0491bc5f697c77092c686038f14f2b60efc34)
 - [fbb81c9470a5356ce7d21f609f0c70bdae9fb8a0](https://github.com/magento/magento2/commit/fbb81c9470a5356ce7d21f609f0c70bdae9fb8a0)
 - [6301ed4e968ca617aef8424a5d322446859ce2e7](https://github.com/magento/magento2/commit/6301ed4e968ca617aef8424a5d322446859ce2e7)

The fix will be available with the upcoming 2.3.0 release.

**Comment by sidolov** [comment]
Hi @angelo983. Thank you for your report. 
The issue has been fixed in magento/magento2#16264 by @chirag-wagento in 2.1-develop branch
Related commit(s):
 - [c9e5239a2786b1169f6f06f61f9589b8078c0865](https://github.com/magento/magento2/commit/c9e5239a2786b1169f6f06f61f9589b8078c0865)

The fix will be available with the upcoming 2.1.15 release.

**Comment by enl** [comment]
Hello @sidolov 

As I can see from @julienanquetil code, it fixes only gift message related IDs. What about billing address form? Is it extracted in separate issue?

**Comment by LiamKarlMitchell** [comment]
Running into this on 2.2.4 just enabled multiple payment methods now checkout does not work.
Could someone please make an extension/hot-fix this as a temporary measure?
Multiple `#billing-new-address-form`

Or tell me which file I can edit to make this fix work?

**Comment by jorisssss** [comment]
I fixed this isue by changing the following setting in Configuration:
Sales - Checkout - Checkout options:
Display Billing Address On : -> Set to "Payment page".

This has solved the issue for me.

**Comment by LiamKarlMitchell** [comment]
Also a problem with #customer-email in authentication-popup.html

**Comment by xceed-will** [comment]
> I fixed this isue by changing the following setting in Configuration:
> Sales - Checkout - Checkout options:
> Display Billing Address On : -> Set to "Payment page".
> 
> This has solved the issue for me.

worked for us jorisssss

**Comment by ghost** [comment]
Hi Guys, just wondering... I've updated to v2.2.6 but I can still see the issue, is this available in that new release or we'll need to wait for v2.3 ? thanks!

**Comment by ishakhsuvarov** [comment]
@bruno-serfe reopening to get it verified once more

**Comment by diazwatson** [comment]
Hi guys, I'll have a look to this issue, could you please assign to me?

**Comment by diazwatson** [comment]
Hi guys, I am looking at this issue in 2.2-develop (2.2.7-dev) and it seems to be fixed.
Can somebody advise what needs to be done in here?

**Comment by magento-engcom-team** [comment]
Hi @diazwatson. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by diazwatson** [comment]
@magento-engcom-team could you give me 2.3-develop instance to deploy test instance on Magento infrastructure please?

**Comment by diazwatson** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @diazwatson. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @diazwatson, here is your [Magento instance](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-13415-2-3-develop/).
Admin access: [http://ec2-34-228-235-121.compute-1.amazonaws.com/i-13415-2-3-develop//admin](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-13415-2-3-develop//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by diazwatson** [comment]
Hey guys I am trying to reproduce this issue with the 2.3-develop instance provided but when I try to add a product to the cart I get a message saying:  
`Your session has expired`

Any idea why this is happening?


If this helps, I have also tested that branch in my local environment and the issue seems to be fixed there.

**Comment by magento-engcom-team** [comment]
Hi @TomashKhamlai. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by TomashKhamlai** [comment]
@magento-engcom-team give me 2.3.0 instance

**Comment by magento-engcom-team** [comment]
Hi @TomashKhamlai. Thank you for your request. I'm working on Magento 2.3.0 instance for you

**Comment by magento-engcom-team** [comment]
Hi @TomashKhamlai, here is your [Magento instance](https://i-13415-2-3-0.instances.magento-community.engineering).
Admin access: [https://i-13415-2-3-0.instances.magento-community.engineering/admin](https://i-13415-2-3-0.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by TomashKhamlai** [comment]
@magento-engcom-team give me 2.2.7 instance

**Comment by magento-engcom-team** [comment]
Hi @TomashKhamlai. Thank you for your request. I'm working on Magento 2.2.7 instance for you

**Comment by magento-engcom-team** [comment]
Hi @TomashKhamlai, here is your [Magento instance](https://i-13415-2-2-7.instances.magento-community.engineering).
Admin access: [https://i-13415-2-2-7.instances.magento-community.engineering/admin](https://i-13415-2-2-7.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by TomashKhamlai** [comment]
On 2.3.0 gift-message-whole-to-giftOptionsCartItem-1 
![screenshot_20181214_153556](https://user-images.githubusercontent.com/11827230/50012247-0fdbe480-ffb6-11e8-9346-ad37e80a5e22.png)

**Comment by TomashKhamlai** [comment]
The same on 2.2.7
Also it looks like it needs some cosmetic repair too.

**Comment by magento-engcom-team** [comment]
@TomashKhamlai Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97159`, `MAGETWO-97160` were created

**Comment by magento-engcom-team** [comment]
Hi @angelo983. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. 

The fix will be available with the upcoming `2.2.10` release.

**Comment by magento-engcom-team** [comment]
Hi @angelo983. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. 

The fix will be available with the upcoming `2.3.3` release.
